packages/design-system/src/components/button.tsx

```diff
@@ -125,25 +125,43 @@ type ButtonProps = {
   // this is a replacement for icon/icon-left/icon-right in Figma
   prefix?: ReactNode;
   suffix?: ReactNode;
+
+  // might be set when <Button> is asChild
+  "data-state"?: string;
 } & Omit<ComponentProps<"button">, "prefix">;
 
 export const Button = forwardRef(
   (
     {
       disabled,
-      state = "auto",
+      state,
       prefix,
       suffix,
       children,
+      "data-state": dataState,
       ...restProps
     }: ButtonProps,
     ref: Ref<HTMLButtonElement>
   ) => {
+    // when button is used as a trigger for something that opens
+    // <SomeTrigger asChild><Button /></SomeTrigger>
+    let finalState = dataState === "open" ? "pressed" : undefined;
```

Comment:
you are mapping it to pressed, but is it supposed to be pressed state? was it discussed with Taylor?

Reply:
Yes, discussed and Taylor documented this in Figma